modules/apps/foundation/roles/roles-admin-web/src/main/java/com/liferay/roles/admin/web/util/RoleAssigneesUtil.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.roles.admin.web.util;
+
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.RoleConstants;
+import com.liferay.portal.kernel.service.RoleLocalServiceUtil;
+import com.liferay.portal.kernel.service.UserGroupRoleLocalServiceUtil;
+import com.liferay.portal.kernel.util.ArrayUtil;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Drew Brokke
+ */
+public class RoleAssigneesUtil {
```

Comment:
I find this class a bit weird.

It seems that it's necessary because you have added many methods to RoleLocalServiceUtil to obtain the count of assignees depending on the role type. I think there should be just one method:

RoleLocalServiceUtil.getAssigneesCount(roleId);

Also, I'm not sure if the fact that a role is implied should make it return 0 in the count. Maybe we should instead try to invoke the method for these roles in the UI.